Library/Formula/graphicsmagick.rb

```diff
@@ -58,15 +59,18 @@ def install
     args << "--with-perl" if build.with? "perl"
     args << "--with-webp=yes" if build.with? "webp"
 
-    if build.with? "quantum-depth-32"
-      quantum_depth = 32
-    elsif build.with? "quantum-depth-16"
+    quantum_depth = { 8 => (build.with? "quantum-depth-8"),
+                     16 => (build.with? "quantum-depth-16"),
+                     32 => (build.with? "quantum-depth-32")}.select{|key, val| val}
+    if quantum_depth.size == 0
       quantum_depth = 16
-    elsif build.with? "quantum-depth-8"
-      quantum_depth = 8
+    elsif quantum_depth.size == 1
+      quantum_depth = quantum_depth.keys[0]
+    else
+      odie "graphicsmagick: --with-quantum-depth-X options are mutually exclusive"
```

Comment:
This part seems a bit convoluted/repetitive and there's really no reason to use a hash. Here's what I would suggest:

```ruby
quantum_depth = [8, 16, 32].select { |n| build.with? "quantum-depth-#{n}" }
if quantum_depth.length > 1
  odie "graphicsmagick: --with-quantum-depth-X options are mutually exclusive"
end
quantum_depth = quantum_depth.first || 16 # user choice or default
```